acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/eth1rpc/EthRpcDownstreamTlsAcceptanceTest.java

```diff
@@ -0,0 +1,188 @@
+import static tech.pegasys.web3signer.dsl.tls.support.CertificateHelpers.populateFingerprintFile;
+
+import tech.pegasys.web3signer.core.config.KeyStoreOptions;
+import tech.pegasys.web3signer.core.config.client.ClientTlsOptions;
+import tech.pegasys.web3signer.dsl.signer.SignerConfigurationBuilder;
+import tech.pegasys.web3signer.dsl.tls.TlsCertificateDefinition;
+import tech.pegasys.web3signer.dsl.tls.client.BasicClientTlsOptions;
+import tech.pegasys.web3signer.dsl.tls.client.BasicKeyStoreOptions;
+import tech.pegasys.web3signer.dsl.tls.support.MockBalanceReporter;
+import tech.pegasys.web3signer.dsl.tls.support.TlsEnabledHttpServerFactory;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.Optional;
+
+import io.vertx.core.http.HttpServer;
+import org.bouncycastle.util.Integers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.web3j.protocol.core.methods.request.Transaction;
+import org.web3j.protocol.exceptions.ClientConnectionException;
+import org.web3j.utils.Convert;
+
+class EthRpcDownstreamTlsAcceptanceTest extends Eth1RpcAcceptanceTestBase {
+
+  private TlsEnabledHttpServerFactory serverFactory;
+
+  @BeforeEach
+  void setup() {
+    serverFactory = new TlsEnabledHttpServerFactory();
+  }
+
+  private void startSigner(
+      final TlsCertificateDefinition presentedCert,
+      final TlsCertificateDefinition expectedWeb3ProviderCert,
+      final int downstreamWeb3Port,
+      final Path workDir)
+      throws IOException, CertificateException, NoSuchAlgorithmException, KeyStoreException {
+
+    final Path clientPasswordFile =
+        Files.writeString(workDir.resolve("clientKeystorePassword"), presentedCert.getPassword());
+
+    final Path fingerPrintFilePath = workDir.resolve("known_servers");
+    final SignerConfigurationBuilder builder =
+        new SignerConfigurationBuilder().withMode("eth1").withDownstreamHttpProxyEnabled(true);
+    final Optional<Integer> downstreamWeb3ServerPort =
+        Optional.of(Integers.valueOf(downstreamWeb3Port));
+
+    populateFingerprintFile(
+        fingerPrintFilePath, expectedWeb3ProviderCert, downstreamWeb3ServerPort);
+
+    final KeyStoreOptions keyStoreOptions =
+        new BasicKeyStoreOptions(presentedCert.getPkcs12File().toPath(), clientPasswordFile);
+    final ClientTlsOptions clientTlsOptions =
+        new BasicClientTlsOptions(keyStoreOptions, Optional.of(fingerPrintFilePath), true);
+    builder.withDownstreamTlsOptions(clientTlsOptions).withDownstreamHttpPort(downstreamWeb3Port);
+
+    startSigner(builder.build());
+  }
+
+  @Test
+  void ethSignerProvidesSpecifiedClientCertificateToDownStreamServer(@TempDir Path workDir)
```

Comment:
is the method name still making reference to ethSigner? should we change that?

Reply:
Yes definitely, good catch. Replaced all the other ethsigner references as well